third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -652,25 +695,28 @@ impl Interpreter {
                 resolver
                     .loader()
                     .check_dependencies_and_charge_gas(
-                        module_store,
+                        resolver.module_store(),
                         data_store,
                         gas_meter,
                         &mut traversal_context.visited,
                         traversal_context.referenced_modules,
                         [(arena_id.address(), arena_id.name())],
+                        resolver.module_storage(),
                     )
                     .map_err(|err| err
                         .to_partial()
                         .append_message_with_separator('.',
                             format!("Failed to charge transitive dependency for {}. Does this module exists?", module_name)
                         ))?;
-                resolver
-                    .loader()
-                    .load_module(&module_name, data_store, module_store)
-                    .map_err(|_| {
-                        PartialVMError::new(StatusCode::FUNCTION_RESOLUTION_FAILURE)
-                            .with_message(format!("Module {} doesn't exist", module_name))
-                    })?;
+
+                if let Loader::V1(loader) = resolver.loader() {
```

Comment:
is this similar reasoning to L621? that comment was very useful, I'd add something similar here (and in general in cases where we do logic specific to one of the V1/V2 variants to explain why not applicable to the other one. (maybe for V2 it will be obvious later since it will be the newer one)

Reply:
done

Reply:
let's extract the ugliness in a helper fn? (seems same as above)